### ISSUE TITLE
After upgrade to adam 0.19, many tests use mostly mismatching read alignments.

### ISSUE THREAD

**Issue by jstjohn** [hammerlab/guacamole#403]
While working on my mutect-like algorithm, and porting tests over from the `SomaticStandardCallerSuite`, I noticed that the [multi-base deletion](https://github.com/hammerlab/guacamole/blob/master/src/test/scala/org/hammerlab/guacamole/commands/SomaticStandardCallerSuite.scala#L178 ) test was not working, while it seemed to work fine for the `SomaticStandardCaller`. After digging in I noticed that the "sum of mismatching base quality score filter" was removing all of the reads from the tumor sample. On digging deeper into the test, I noticed that the test was set up to put the normal/tumor read on a reference sequence that didn't match the normal read sequence. I was able to get that test passing by adding another contig that matched the normal read sequence, and use that instead.

I am a little concerned that this might be a more global problem. Do any other methods in guacamole make use of filters based on mismatching bases? If so any `negative` tests need to be examined carefully to make sure this recent transition away from user defined MD tags is not causing any other unexpected issues.

In the case of the SomaticStandardCaller I think that bogus alignment is no problem. Basically it is an alignment where all bases either mismatch, or based on the cigar are marked as deleted, so I think this test is still doing something useful in that case. 

As a user though it threw me off, and I am not sure that my other passing negative tests in MuTect are because a bogus alignment is getting filtered in my base caller function, or if it is actually negative. I'll have to be careful :)

**Comment by timodonnell** [comment]
Great catch, thanks @jstjohn 

The recent remove MD tags change involved lots of fiddling with tests, so it's disappointing but not too surprising that I missed some of this up.

Is your fix (the addition of a new contig with the correct reference sequence) available somewhere?

**Comment by jstjohn** [comment]
Sure here are the relevant updates I made to https://github.com/drivergroup/guacamole/blob/mutectImplementationWork/src/test/scala/org/hammerlab/guacamole/commands/SomaticMutectLikeCallerSuite.scala. Just to be clear I have not thoroughly gone through the other tests in that class yet. Everything passes but I am not sure the negatives are actually passing for the right reason in my case.

```
  def simpleReference = TestUtil.makeReference(sc, Seq(
    ("chr1", 0, "TCGATCGACG"),
    ("chr2", 0, "TCGAAGCTTCG"),
    ("chr3", 10, "TCGAATCGATCGATCGA"),
    ("chr4", 0, "TCGAAGCTTCGAAGCT")))
```

```
  sparkTest("multiple-base deletion") {
    val normalReads = Seq(
      TestUtil.makeRead("TCGAAGCTTCGAAGCT", "16M", 0, "chr4"),
      TestUtil.makeRead("TCGAAGCTTCGAAGCT", "16M", 0, "chr4"),
      TestUtil.makeRead("TCGAAGCTTCGAAGCT", "16M", 0, "chr4")
    )
    //TCGAAGCTTCG
    val normalPileup = Pileup(normalReads, "chr4", 4, referenceContigSequence = simpleReference.getContig("chr4"))

    val tumorReads = Seq(
      TestUtil.makeRead("TCGAAAAGCT", "5M6D5M", 0, "chr4"), // md tag: "5^GCTTCG5"
      TestUtil.makeRead("TCGAAAAGCT", "5M6D5M", 0, "chr4"),
      TestUtil.makeRead("TCGAAAAGCT", "5M6D5M", 0, "chr4")
    )
    val tumorPileup = Pileup(tumorReads, "chr4", 4, referenceContigSequence = simpleReference.getContig("chr4"))
```

**Comment by timodonnell** [comment]
This was good to take a closer look at. I think I agree @jstjohn that this is not actually an issue for the `SomaticStandardCallerSuite` (nor for other suites that I took a quick look at). I have a fix for the narrow issue in #405. Definitely interested if you spot any similar mistakes anywhere else. Thanks!
